omod-1.9/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_9/ConceptReferenceTermResource1_9.java

```diff
@@ -163,9 +164,14 @@ protected PageableResult doSearch(RequestContext context) {
 		String query = context.getParameter("q");
 		if (query == null)
 			return new EmptySearchResult();
-		List<ConceptReferenceTerm> terms = cs.getConceptReferenceTerms(query, null, context.getStartIndex(),
+        String sourceUuid = context.getParameter("source");

```

Comment:
@dkayiwa @wluyima now I got it, when user specifies source, he can't use `q`, but other SearchHandler parameters, like `code` and `name`. When query is specified, controller goes straight to `doSearch` method. It makes sense, but should be emphasized in generated docs, that using `q` and other search parameters are mutually exclusive. Either way, I think that makes this issue closed.